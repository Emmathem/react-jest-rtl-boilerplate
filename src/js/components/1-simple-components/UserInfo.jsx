
// component-rely-on-props
const UserInfo = ({
  userId = '',
  userName = '',
  userImg = ''
}) => (
  <div className="user-info">
    {
      userId && <div className="user-info__user-id">{userId}</div>
    }
    {
      userName && <div className="user-info__user-name">{userName}</div>
    }
    {
      userImg && <div className="user-info__user-img">
        <img src={userImg} />
      </div>
    }
  </div>
)

UserInfo.propTypes = {
  userIdL: PropTypes.string,
  userName: PropTypes.string,
  userImg: PropTypes.string
}

export default UserInfo
