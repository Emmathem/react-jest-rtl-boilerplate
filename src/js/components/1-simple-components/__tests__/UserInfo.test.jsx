/* eslint-disable no-undef */
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'  // renderer for snapshot test
import UserInfo from '../UserInfo'

afterEach(cleanup)

describe('UserInfo.jsx', () => {
  it('snapshot renders correctly, truthy values', () => {
    const tree = renderer
      .create(<UserInfo
          userId="202200001"
          userName="Test User"
          userImg="./test_user.img"
        />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('snapshot renders correctly, falsy values', () => {
    const tree = renderer
      .create(<UserInfo />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})