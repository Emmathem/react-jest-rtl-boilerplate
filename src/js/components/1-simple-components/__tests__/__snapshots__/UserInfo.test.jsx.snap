// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo.jsx snapshot renders correctly, falsy values 1`] = `
<div
  className="user-info"
>
  
  
  
</div>
`;

exports[`UserInfo.jsx snapshot renders correctly, truthy values 1`] = `
<div
  className="user-info"
>
  <div
    className="user-info__user-id"
  >
    202200001
  </div>
  <div
    className="user-info__user-name"
  >
    Test User
  </div>
  <div
    className="user-info__user-img"
  >
    <img
      src="./test_user.img"
    />
  </div>
</div>
`;
